Migrate Mono APIs to CoreCLR-compatible APIs

MiroBrodlovaUnityGit opened the pull request:
## Purpose of PR

This PR replaces all instances of analyzer-highlighted `Mono` API calls with `CoreCLR`-compatible API calls in this repo, and is part of Epic [SCP-1555](https://jira.unity3d.com/browse/SCP-1555).

## Further Context

- Unity is steadily moving away from `Mono` to using `CoreCLR`. 
- Epic [SCP-1555](https://jira.unity3d.com/browse/SCP-1555) tracks this work.
- For more information, please visit [#code-reload-safe-api-packages-migration](https://unity.enterprise.slack.com/archives/C09N2RM0L4D) 🙂 

### JIRA-ticket
- [SCP-1555](https://jira.unity3d.com/browse/SCP-1555)

## Testing
Running the default job-suite. Please share if there are additional steps needed!

todi1856 reviewed (approved):
Any reason why other files were changed, do you use some kind of formatter ?

MiroBrodlovaUnityGit commented:
> Any reason why other files were changed, do you use some kind of formatter ?

Hi Tomas! 
Yes, my bad, it wasn't my intention to change these files. I was creating multiple PRs across (with almost identical changes) a few repos so was using Claude to push my change automatically into the draft PR, and for some reason it added these additional file-"changes". I was a little skeptical the AI may do some commit-shenanigans so I limited it to only draft PRs so I could review and remove things like this before opening it up to be a proper PR.

Thanks for looking/alerting me! 
Will remove those commit-changes (they seem to actually not do anything, as the contents between the changes are the same, so it's extra strange it happened) and verify only needed changes are included. Thanks. 

Will re-ask for a review once all changes are cleaned up :) so you can be sure it looks good!

MiroBrodlovaUnityGit committed:
Migrate AppDomain.GetAssemblies() to CurrentAssemblies.GetLoadedAssemblies()

- AndroidBridge.cs: Add conditional compilation for CoreCLR compatibility
- Added using UnityEngine.Assemblies with conditional directive
- Maintains backward compatibility with older Unity versions
- Updated CHANGELOG.md with migration note

JIRA: SCP-1555
